src/scripting/lua_widget_attributes.cpp

```diff
@@ -0,0 +1,527 @@
+{
+	if(w.get_selected_row() >= w.get_item_count()) {
+		throw std::invalid_argument("widget has no selected item");
+	}
+	return w.get_row_grid(w.get_selected_row());
+}
+*/
+
+WIDGET_SETTER("value_compat,selected_index", int, gui2::listbox)
+{
+	w.select_row(value - 1);
+}
+
+WIDGET_GETTER("value_compat,selected_index", int, gui2::multi_page)
+{
+	return w.get_selected_page() + 1;
+}
+
+WIDGET_SETTER("value_compat,selected_index", int, gui2::multi_page)
+{
+	w.select_page(value -1);
+}
+
+WIDGET_GETTER("value_compat,selected_index", int, gui2::stacked_widget)
+{
+	return w.current_layer() + 1;
+}
+
+WIDGET_SETTER("value_compat,selected_index", int, gui2::stacked_widget)
+{
+	w.select_layer(value - 1);
+}
+
+WIDGET_GETTER("selected_index", int, gui2::selectable_item)
+{
+	return w.get_value() + 1;
+}
+
+WIDGET_SETTER("selected_index", int, gui2::selectable_item)
+{
+	if(value > int(w.num_states())) {
+		throw std::invalid_argument("invalid index");
+	}
+	w.set_value(value + 1);
+}
+
+WIDGET_GETTER("value_compat,selected", bool, gui2::selectable_item)
+{
+	if(w.num_states() == 2) {
+		return w.get_value_bool();
+	}
+	throw std::invalid_argument("invalid widget");
+}
+
+WIDGET_SETTER("value_compat,selected", bool, gui2::selectable_item)
+{
+	w.set_value_bool(value);
+}
+
+WIDGET_GETTER("value_compat,text", std::string, gui2::text_box)
+{
+	return w.get_value();
+}
+
+WIDGET_SETTER("value_compat,text", std::string, gui2::text_box)
+{
+	w.set_value(value);
+}
+
+WIDGET_GETTER("value_compat,value", int, gui2::slider)
+{
+	return w.get_value();
+}
+
+WIDGET_SETTER("value_compat,value", int, gui2::slider)
+{
+	w.set_value(value);
+}
+
+WIDGET_GETTER("value_compat,percentage", int, gui2::progress_bar)
+{
+	return w.get_percentage();
+}
+
+WIDGET_SETTER("value_compat,percentage", int, gui2::progress_bar)
+{
+	w.set_percentage(value);
+}
+
+WIDGET_GETTER("value_compat,selected_item_path", std::vector<int>, gui2::tree_view)
+{
+	auto res = w.selected_item()->describe_path();
+	for(int& a : res) { ++a;}
+	return res;
+}
+
+WIDGET_GETTER("path", std::vector<int>, gui2::tree_view_node)
+{
+	auto res = w.describe_path();
+	for(int& a : res) { ++a;}
+	return res;
+}
+
+WIDGET_SETTER("value_compat,unfolded", bool, gui2::tree_view_node)
+{
+	if(value) {
+		w.unfold();
+	} else {
+		w.fold();
+	}
+}
+
+WIDGET_SETTER("value_compat,unit", lua_index_raw, gui2::unit_preview_pane)
+{
+	if(const unit_type* ut = luaW_tounittype(L, value.index)) {
+		w.set_displayed_type(*ut);
+	} else if(unit* u = luaW_tounit(L, value.index)) {
+		w.set_displayed_unit(*u);
+	} else {
+		luaW_type_error(L, value.index, "unit or unit type");
+	}
+}
+
+WIDGET_GETTER("item_count", int, gui2::multi_page)
+{
+	return w.get_page_count();
+}
+
+WIDGET_SETTER("use_markup", bool, gui2::styled_widget)
+{
+	w.set_use_markup(value);
+}
+
+//TODO: while i think this shortcut is useful, i'm not that happy about
+//      the name since  it changes 'label' and not 'text', the first one
```

Comment:
You could just change it to `marked_up_label` then? Or for something a little shorter, maybe `rich_label`?

Reply:
hmm bit isnt it the text on the label that is marked up and not the label ?

Reply:
I'm not sure, I was just responding to the content of the comment. It was just a random suggestion; if it doesn't make sense, don't use it?

I'd think of the label and the text of the label to be the same thing, personally. Not sure if that makes sense in GUI2.

Reply:
> I'd think of the label and the text of the label to be the same thing, personally. Not sure if that makes sense in GUI2.

Yes me too, im not sure whether we can though, if i undestood gui2 correctly widgets like  for example textboxes can have a label attached to them. And then these widgets do really have a  "label" and a "text"